rust_cli/src/commands/idl.rs

```diff
@@ -0,0 +1,512 @@
+            &[ComputeBudgetInstruction::set_compute_unit_price(priority_fees_per_cu), ComputeBudgetInstruction::set_compute_unit_limit(5000),create_account_ix, create_buffer],
+            Some(&signer.pubkey()),
+            &[signer, &buffer_keypair],
+            recent_blockhash,
+        );
+    let signature = rpc_client
+        .send_and_confirm_transaction_with_spinner_and_commitment(&transaction, CommitmentConfig::confirmed())
+        .map_err(|e| anyhow!("Failed to send transaction: {}", e))?;
+    println!("Buffer account created successfully!");
+    println!("Signature: {}", signature);
+    println!("Buffer Account: {}", buffer_keypair.pubkey());
+    Ok((compressed_data, buffer_keypair))
+} 
+
+
+fn write_buffer(
+    compressed_data: Vec<u8>, 
+    buffer_keypair: &Keypair, 
+    signer: &Keypair, rpc_client: &solana_client::rpc_client::RpcClient, 
+    priority_fees_per_cu: u64
+) -> Result<(), anyhow::Error> {
+    let mut offset = 0;
+    while offset < compressed_data.len() {
+        let chunk_end = std::cmp::min(offset + CHUNK_SIZE as usize, compressed_data.len());
+        let chunk = &compressed_data[offset..chunk_end];
+    
+        let write_buffer = WriteBuffer {
+            buffer: buffer_keypair.pubkey(),
+            signer: signer.pubkey(),
+            system_program: solana_sdk::system_program::ID,
+        };
+    
+        let write_args = WriteBufferInstructionArgs {
+            idl_data: chunk.to_vec(),
+        };
+    
+        let write_ix = write_buffer.instruction(write_args);
+    
+        let recent_blockhash = rpc_client
+            .get_latest_blockhash()
+            .map_err(|e| anyhow!("Failed to get write buffer blockhash: {}", e))?;
+    
+        let write_transaction = Transaction::new_signed_with_payer(
+            &[
+                ComputeBudgetInstruction::set_compute_unit_price(priority_fees_per_cu),
+                ComputeBudgetInstruction::set_compute_unit_limit(3000),
+                write_ix
+            ],
+            Some(&signer.pubkey()),
+            &[signer],
+            recent_blockhash,
+        );
+    
+        let write_signature = rpc_client
+            .send_and_confirm_transaction_with_spinner_and_commitment(
+                &write_transaction,
+                CommitmentConfig::confirmed()
+            )
+            .map_err(|e| anyhow!("Failed to send write buffer transaction: {}", e))?;
+    
+        println!("Wrote chunk {} to {}, signature: {}", offset, chunk_end, write_signature);
+    
+        offset = chunk_end;
+    }
+    println!("All buffer chunks written successfully!");
+
+    Ok(())
+}
+
+fn set_and_close_buffer(
+    rpc_client: solana_client::rpc_client::RpcClient, 
+    idl_address: Pubkey, 
+    buffer_keypair: Keypair, 
+    priority_fees_per_cu: u64, 
+    signer: Keypair, 
+    program_pubkey: Pubkey,
+    seed: &str,
+) -> Result<(), anyhow::Error> {
+    // Try to get account info with retries
+    let mut attempts = 0;
```

Comment:
A bit strange that here we use a retry but on all the other places not. Maybe add a function getAccountWithRetry and use it everywhere?